app/models/application_record.rb

```diff
@@ -26,13 +30,16 @@ def attributes_print
   def self.sanitize_for_search(term, **cols)
```

Comment:
The original behaviour of this method was to take a search parameter and list of columns, and turn them into a search against a _single_ fulltext index, i.e.

```ruby
sanitize_for_search 'query', posts: :body
# => "MATCH (`posts`.`body`) AGAINST ('query' IN BOOLEAN MODE)"
```

Do I understand right that what you're trying to do is make this return multiple `MATCH ... AGAINST` clauses for multiple fulltext indexes? At the moment, I don't think this does what you want it to. If you pass in an array of columns, you get the array of clauses:

```ruby
sanitize_for_search 'query', posts: [:body, :title]
# => [" MATCH `posts`.`body` AGAINST ('query' IN BOOLEAN MODE)", " MATCH `posts`.`title` AGAINST ('query' IN BOOLEAN MODE)"]
```

If you only pass a single column, you just get the column's name, no MATCH clause at all:

```ruby
sanitize_for_search 'query', posts: :body
# => "`posts`.`body`"
```

What's probably an easier way to do it (if I understand what you want correctly) is:

 * keep the purpose of the `cols = cols.map` block the same, i.e. map unsanitized column names to sanitized column names; then
 * return a map of that, which turns it into the MATCH clauses.

Something like this:

```ruby
cols = cols.map do |table_name, column_or_array|
  if column_or_array.is_a?(Array)
    column_or_array.map { |column_name| "#{sanitize_name table_name}.#{sanitize_name column_name}" }
  else
    "#{sanitize_name table_name}.#{sanitize_name column_or_array}"
  end
end.flatten

cols.map do |sanitized_name|
  ActiveRecord::Base.sanitize_sql(["MATCH (#{sanitized_name}) AGAINST (? IN BOOLEAN MODE)", term])
end
```

Reply:
> Do I understand right that what you're trying to do is make this return multiple MATCH ... AGAINST clauses for multiple fulltext indexes?
Yup exactly what I was trying to do. Ran your suggestion and it worked like a charm! 

Would there be a case where we pass in a single column? The methods are first being called here https://github.com/codidact/qpixel/blob/develop/app/models/post.rb#L59 and it looks like it will only ever look at the columns that we tell it. If there is, then the way I am trying to do the search weight would break (not that it was a good way to do it anyways 😛)